src/pyOpenMS/pyopenms/_dataframes.py

```
@@ -652,87 +652,34 @@ def _add_meta_values(df: _pd.DataFrame, object: any) -> _pd.DataFrame:
         try:
-            v = int(v)
-            dtype = int
-        except ValueError:
-            try:
-                v = float(v)
-                dtype = 'double'
-            except ValueError:
-                dtype = f'U{len(v)}'
-        
-        df[k.decode()] = _np.full(df.shape[0], v, dtype=_np.dtype(dtype))
+            df[k_str] = np.full(cnt, v, dtype=dtype)
```

Ruff (0.8.2)

Undefined name `np`

(F821)